engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -1516,6 +1531,20 @@ public void run()
     }
   }
 
+  /*
+   * Generates groupId for operator error event.
+   * The events generated by downstream operators redeployment should use this groupId
+   */
+  private long getUndeployGroupId(int[] operators)
```

Comment:
In the current design the container generates undeployGroupId for all the operators getting undeployed which is communicated to the StrAM thru the heartbeat and the StrAM then communicates this same thru the heartbeat response and then to the other container thru the deploy command. Why can't we simplify this where all the groupId generation and assignment happens only in the StrAM and since the StrAM anyway needs to send the undeploy and deploy commands to the containers they can include the groupId. Couple of advantages is that it also covers the case of a container suddenly dying (without getting a chance to call reportError or heartbeat to the StrAM) and a lot of code/logic here then becomes redundant. If the logic and the changes are simplified then it is easier to review the PR and also approve.